### PR TITLE
API

### PR DESCRIPTION
**Pull request by TheDoplarEffect**
The original PR was created from a branch and not directly from main.

causing lots of changed files all over the system.  

So this is a new, slick, updated PR.

One file, one class.  I have chosen this structure with the intent that in components that need the API, they can create a new instance of the class, and get the Data they need with a method.

If more info or a different type of info are needed, only the API file needs changing or another method to be added.

### CODE REVIEW

File: src/components/API.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import axios from "axios";
+
+export type DataResponce = {
+  id: number;
+  poster_path: string;
+};
+
+export class API {
```
Comment (dnmct) [inline src/components/API.tsx:8]: use a loader instead

**Review by dnmct — CHANGES_REQUESTED** [review]

**Comment by TheDoplarEffect** [comment]
Functions will now just return the data.  This way different Loaders access the funcitons for the specific Data they need.

File: src/components/API.tsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import axios from "axios";
+import { Movie } from "../types/api";
+
+export class API {
```
Comment (NoHara42) [inline src/components/API.tsx:4]: you know classes aren't real right? :p 
You could export a const called API, which has functions as properties of the objects.

**Review by NoHara42 — APPROVED** [review]
amazeballs

**Pull request merged**